src/v/security/audit/audit_log_manager.h

```diff
@@ -126,6 +128,20 @@ class audit_log_manager
           make_api_activity_event(std::forward<Args>(args)..., {})));
     }
 
+    template<typename... Args>
+    bool enqueue_app_lifecycle_event(Args... args) {
```

Comment:
the issue with using the variable template arguments here is that `event<activity_id>` is a separate type from `event<activity_id, activity_id>`, so we're generating more code. you could use an initializer list or take a vector.

```
void func2( std::initializer_list<activity_id> list )
{
    for( auto elem : list )
    {
    }
}
```